### PR TITLE
ADX : Added `adxOnBehalfOf` feature gate

### PR DESCRIPTION
**Pull request by yaelleC**
**What this PR does / why we need it**:

We ran into a few issues with the OBO feature in ADX (in beta atow). 
This PR provides a feature gate `adxOnBehalfOf` to allow us to hide the toggle in ADX until issues are resolved.

First part of: https://github.com/grafana/azure-data-explorer-datasource/issues/359 

**Which issue(s) this PR fixes**:

First part of: https://github.com/grafana/azure-data-explorer-datasource/issues/359 

**Special notes for your reviewer**:

Next step is to use this feature gate in ADX code to hide the toggle and disable OBO if feature disabled.

### CODE REVIEW

File: conf/defaults.ini [hunk 1]
```diff
@@ -1189,6 +1189,9 @@ cloudWatchDynamicLabels = true
 # New expandable navigation
 newNavigation = true
 
+# On-Behalf-Of functionality for ADX data source
+adxOnBehalfOf = true
```
Comment (joshhunt) [inline conf/defaults.ini:1193]: It's intended that the feature flag is enabled by default?

Comment (yaelleC) [reply conf/defaults.ini:1193]: It was, but we are re-thinking it now 😉 
Full discussion here: https://raintank-corp.slack.com/archives/C01LVB0CYTW/p1653360314223909

I will remove it now

File: pkg/services/featuremgmt/registry.go [hunk 2]
```diff
@@ -249,5 +249,10 @@ var (
 			State:           FeatureStateAlpha,
 			RequiresRestart: true,
 		},
+		{
+			Name:        "adxOnBehalfOf",
+			Description: "Enable the beta feature On-Behalf-Of in the ADX data source",
+			State:       FeatureStateBeta,
+		},
```
[the remark this answers is not in the document]
Comment (yaelleC) [reply pkg/services/featuremgmt/registry.go:256]: Yes, I was starting to wonder the same thing...

[the remark this answers is not in the document]
Comment (kostrse) [reply pkg/services/featuremgmt/registry.go:256]: We probably could use `azsettings.AzureSettings` for that?

**Comment by yaelleC** [comment]
Closing as we will only use frontend changes in ADX - thus this PR is not needed anymore

**Pull request closed without merging**